translib/transformer/xfmr_mclag.go

```diff
@@ -0,0 +1,266 @@
+//                                                                            //
+////////////////////////////////////////////////////////////////////////////////
+
+package transformer
+
+import (
+	log "github.com/golang/glog"
+	"github.com/openconfig/ygot/ygot"
+	"strconv"
+	"strings"
+	"translib/db"
+	"translib/ocbinds"
+)
+
+func init() {
+	XlateFuncBind("YangToDb_mclag_domainid_fld_xfmr", YangToDb_mclag_domainid_fld_xfmr)
+	XlateFuncBind("DbToYang_mclag_domainid_fld_xfmr", DbToYang_mclag_domainid_fld_xfmr)
+	XlateFuncBind("YangToDb_mclag_vlan_name_fld_xfmr", YangToDb_mclag_vlan_name_fld_xfmr)
+	XlateFuncBind("DbToYang_mclag_vlan_name_fld_xfmr", DbToYang_mclag_vlan_name_fld_xfmr)
+	XlateFuncBind("YangToDb_mclag_interface_subtree_xfmr", YangToDb_mclag_interface_subtree_xfmr)
+	XlateFuncBind("DbToYang_mclag_interface_subtree_xfmr", DbToYang_mclag_interface_subtree_xfmr)
+
+	XlateFuncBind("DbToYang_mclag_domain_oper_status_fld_xfmr", DbToYang_mclag_domain_oper_status_fld_xfmr)
+	XlateFuncBind("DbToYang_mclag_domain_role_fld_xfmr", DbToYang_mclag_domain_role_fld_xfmr)
+	XlateFuncBind("DbToYang_mclag_domain_system_mac_fld_xfmr", DbToYang_mclag_domain_system_mac_fld_xfmr)
+}
+
+var YangToDb_mclag_domainid_fld_xfmr FieldXfmrYangToDb = func(inParams XfmrParams) (map[string]string, error) {
+	res_map := make(map[string]string)
+	var err error
+	log.Info("YangToDb_mclag_domainid_fld_xfmr: ", inParams.key)
+
+	return res_map, err
+}
+
+var DbToYang_mclag_domainid_fld_xfmr FieldXfmrDbtoYang = func(inParams XfmrParams) (map[string]interface{}, error) {
+	var err error
+	result := make(map[string]interface{})
+	log.Info("DbToYang_mclag_domainid_fld_xfmr: ", inParams.key)
+	result["domain-id"], _ = strconv.ParseUint(inParams.key, 10, 32)
+
+	return result, err
+}
+
+var YangToDb_mclag_vlan_name_fld_xfmr FieldXfmrYangToDb = func(inParams XfmrParams) (map[string]string, error) {
+	res_map := make(map[string]string)
+	var err error
+	log.Info("YangToDb_mclag_vlan_name_fld_xfmr: ", inParams.key)
+
+	return res_map, err
+}
+
+var DbToYang_mclag_vlan_name_fld_xfmr FieldXfmrDbtoYang = func(inParams XfmrParams) (map[string]interface{}, error) {
+	var err error
+	result := make(map[string]interface{})
+	log.Info("DbToYang_mclag_vlan_name_fld_xfmr: ", inParams.key)
+	result["name"] = inParams.key
+
+	return result, err
+}
+
+var DbToYang_mclag_domain_oper_status_fld_xfmr FieldXfmrDbtoYang = func(inParams XfmrParams) (map[string]interface{}, error) {
+	var err error
+	result := make(map[string]interface{})
+	log.Infof("DbToYang_mclag_domain_oper_status_fld_xfmr --> key: %v", inParams.key)
+
+	stDb := inParams.dbs[db.StateDB]
+	mclagEntry, _ := stDb.GetEntry(&db.TableSpec{Name: "MCLAG_TABLE"}, db.Key{Comp: []string{inParams.key}})
+	operStatus := mclagEntry.Get("oper_status")
+	if operStatus == "up" {
+		result["oper-status"], _ = ygot.EnumName(ocbinds.OpenconfigMclag_Mclag_MclagDomains_MclagDomain_State_OperStatus_OPER_UP)
```

Comment:
Enums like these "OpenconfigMclag_Mclag_MclagDomains_MclagDomain_State_OperStatus_OPER_UP" may not compile without openconfig-mclag.yang, since ocbinds.go will not generate enums for these. Also in all other similar places like these, we will get compilation error. If openconfig-mclag.yang is going to come as part of other PR, then these code can come later as well. Thoughts?

Reply:
will update the oc yang part of this PR. Thanks

Reply:
OK then will wait for those changes, before approving this PR

Reply:
rathnasabapathyv, i have updated oc yang along with this PR please go ahead and look at it and approve it. thanks